sdk/python/tests/dsl/pipeline_param_tests.py

```diff
@@ -51,4 +52,14 @@ def test_extract_pipelineparam(self):
     params = _extract_pipelineparams(payload)
     self.assertListEqual([p1, p2, p3], params)
 
-  #TODO: add more unit tests to cover real type instances
\ No newline at end of file
+  def test_extract_pipelineparam_with_types(self):
+    p1 = PipelineParam(name='param1', op_name='op1', param_type=TypeMeta(name='customized_type_a', properties={'property_a': 'value_a'}))
+    p2 = PipelineParam(name='param2', param_type=TypeMeta(name='customized_type_b'))
+    p3 = PipelineParam(name='param3', value='value3', param_type=TypeMeta(name='customized_type_c', properties={'property_c': 'value_c'}))
+    stuff_chars = ' between '
+    payload = str(p1) + stuff_chars + str(p2) + stuff_chars + str(p3)
+    params = _extract_pipelineparams(payload)
+    self.assertListEqual([p1, p2, p3], params)
+    payload = [str(p1) + stuff_chars + str(p2), str(p2) + stuff_chars + str(p3)]
```

Comment:
nit: Adding a comment here that this test verifies that the method will dedup parameters if it appears in is several places.

Reply:
Good point. Done